src/utils/mergeTree.js

```diff
@@ -212,37 +233,53 @@ async function mergeBlobs({
   const type = 'blob'
   // Compute the new mode.
   // Since there are ONLY two valid blob modes ('100755' and '100644') it boils down to this
-  const mode =
-    (await base.mode()) === (await ours.mode())
-      ? await theirs.mode()
-      : await ours.mode()
-  // The trivial case: nothing to merge except maybe mode
-  if ((await ours.oid()) === (await theirs.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
-  }
-  // if only one side made oid changes, return that side's oid
-  if ((await ours.oid()) === (await base.oid())) {
-    return { mode, path, oid: await theirs.oid(), type }
-  }
-  if ((await theirs.oid()) === (await base.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
+  let ourMode = await ours.mode();
+  if (base !== null) {
+    const mode =
+      (await base.mode()) === (ourMode)
+        ? await theirs.mode()
+        : ourMode
+    // The trivial case: nothing to merge except maybe mode
+    if ((await ours.oid()) === (await theirs.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
+    // if only one side made oid changes, return that side's oid
+    if ((await ours.oid()) === (await base.oid())) {
+      return { mode, path, oid: await theirs.oid(), type }
+    }
+    if ((await theirs.oid()) === (await base.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
   }
   // if both sides made changes do a merge
+  let baseContent;
+  try {
+    baseContent = Buffer.from(await base.content()).toString('utf8');
+  } catch(error) {
+    baseContent = Buffer.from("").toString('utf8');
+  }
   const { mergedText, cleanMerge } = mergeFile({
     ourContent: Buffer.from(await ours.content()).toString('utf8'),
-    baseContent: Buffer.from(await base.content()).toString('utf8'),
+    baseContent,
     theirContent: Buffer.from(await theirs.content()).toString('utf8'),
     ourName,
     theirName,
     baseName,
     format,
     markerSize,
   })
+
   let awaitedMergedText = mergedText
   if (!cleanMerge) {
     // all other types of conflicts fail
     try {
-      awaitedMergedText = await asyncMergeConflictCallback(mergedText, base._fullpath)
+      let fullpath = ""
+      try {
+        fullpath = base._fullpath
```

Comment:
How is it possible for this assignment to result in a thrown exception?

Reply:
The case that I introduced/am handling with these code changes is one where base === null

So any accessors or function calls where it is null will fail, so I added catch statements to prop up that case